skimage/transform/histogram_matching.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+
+def _match_cumulative_cdf(source, template):
+    """
+    Return modified source array so that the cumulative density function of
+    its values matches the cumulative density function of the template.
+    """
+    src_values, src_unique_indices, src_counts = np.unique(source.ravel(),
+                                                           return_inverse=True,
+                                                           return_counts=True)
+    tmpl_values, tmpl_counts = np.unique(template.ravel(), return_counts=True)
+
+    # calculate normalized quantiles for each array
+    src_quantiles = np.cumsum(src_counts).astype(np.float64)
+    src_quantiles /= src_quantiles[-1]
+
+    tmpl_quantiles = np.cumsum(tmpl_counts).astype(np.float64)
+    tmpl_quantiles /= tmpl_quantiles[-1]
+
+    interp_a_values = np.interp(src_quantiles, tmpl_quantiles, tmpl_values)
+    return interp_a_values[src_unique_indices].reshape(source.shape)
+
+
+
+def match_histograms(image, reference):
+    """Adjust an image so that its cumulative histogram matches that of another.
+
+    The adjustment is applied separately for each channel.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image. Can be gray-scale or in color
+    reference : ndarray
+        Image to match histogram of. Must have the same number of channels as
+        image.
+
+    Returns
+    -------
+    matched : ndarray
+        Transformed output image.
+
+    Raises
+    ------
+    ValueError
+        Thrown when the number of channels in the input image and the reference
+        differ.
+
+    References
+    ----------
+    .. [1] http://paulbourke.net/miscellaneous/equalisation/
+
+    """
+    shape = image.shape
+    image_dtype = image.dtype
+
+    if image.ndim != reference.ndim:
+        raise ValueError('Image and reference must have the same number of channels.')
+
+    if image.ndim == 2:
```

Comment:
Can we add a multichannel argument so that this can be applied to nd images? Literally nothing else needs to change in the code, only the API for multi-channel images.